### PR TITLE
Specific reduction cases performance testing

### PR DESCRIPTION
**Pull request by KulikovNikita**

### CODE REVIEW

File: cpp/oneapi/dal/backend/primitives/reduction/test/reduction_rm_cw_perf_dpc.cpp [hunk 1]
```diff
@@ -51,6 +51,15 @@ class reduction_rm_test_uniform : public te::float_algo_fixture<std::tuple_eleme
         CAPTURE(width_, stride_, height_);
     }
 
+    void generate_special() {
+        width_ = GENERATE(28, 960, 2000, 3072);
+        height_ = GENERATE(1024, 4096);
+        stride_ = height_;
+        SKIP_IF(width_ > stride_);
```
Comment (rlnx) [inline cpp/oneapi/dal/backend/primitives/reduction/test/reduction_rm_cw_perf_dpc.cpp:58]: This works only in TEST: `SKIP_IF` is just a macro that expands to `if (expr) return`. Suggest to add `width_ > stride_` check to `should_be_skipped`.

Comment (KulikovNikita) [reply cpp/oneapi/dal/backend/primitives/reduction/test/reduction_rm_cw_perf_dpc.cpp]: Fixed

**Review by rlnx — APPROVED** [review]

File: cpp/oneapi/dal/backend/primitives/reduction/test/reduction_rm_cw_perf_dpc.cpp [hunk 2]
```diff
@@ -46,8 +46,13 @@ class reduction_rm_test_uniform : public te::float_algo_fixture<std::tuple_eleme
         width_ = GENERATE(16, 128, 1024);
         stride_ = GENERATE(16, 128, 1024);
         height_ = GENERATE(16, 128, 1024, 16384, 32768);
-        SKIP_IF(width_ > stride_);
-        REQUIRE(width_ <= stride_);
+        CAPTURE(width_, stride_, height_);
+    }
+
+    void generate_special() {
```
Comment (michael-smirnov) [inline cpp/oneapi/dal/backend/primitives/reduction/test/reduction_rm_cw_perf_dpc.cpp:52]: `generate_special` - a meaningless name for me. Can we provide more detailed naming?

**Review by michael-smirnov — APPROVED** [review]
Minor comments

File: cpp/oneapi/dal/backend/primitives/reduction/test/reduction_rm_cw_perf_dpc.cpp (hunk 2 above)
Comment (KulikovNikita) [reply cpp/oneapi/dal/backend/primitives/reduction/test/reduction_rm_cw_perf_dpc.cpp:52]: Tbh I can not imagine more meaningful name for this function

**Pull request merged**